src/messaging/ExchangeTransport.h

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines ExchangeTransport class. The object of this
+ *      class can be used by CHIP protocols to send and receive messages.
+ */
+
+#pragma once
+
+#include <transport/SecureSessionMgr.h>
+
+namespace chip {
+namespace Messaging {
+
+class ReliableMessageMgr;
+class ReliableMessageContext;
+
+class ExchangeTransport
```

Comment:
It looks to me this class offload part of transmit/re-transmit functionality from ExchangeMgr, right? Then it should be accessory class to ExchangeMgr, not part of ExchangeContext which has 16 instance by default.

Reply:
Discussed this in the slack channel. Also the latest commit updates the name to remove confusion.